vm-superio/src/serial.rs

```diff
@@ -289,6 +291,18 @@ pub enum Error<E> {
     FullFifo,
 }
 
+impl<E: fmt::Display> fmt::Display for Error<E> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::Trigger(e) => write!(f, "Failed to trigger interrupt: {}", e),
+            Error::IOError(e) => write!(f, "Couldn't write/flush to the given destination: {}", e),
+            Error::FullFifo => write!(f, "No space left in FIFO"),
+        }
+    }
+}
+
+impl StdError for Error<io::Error> {}
```

Comment:
This should also be generic over `E`.
```suggestion
impl<E: StdError> StdError for Error<E> {}
```

Reply:
got it!